File: doc-examples/jdbc-sqlite/src/main/java/example/Book.java
package example;

import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import org.jspecify.annotations.Nullable;

@MappedEntity
public record Book(@Id @GeneratedValue @Nullable Long id, String title) {
}

File: doc-examples/r2dbc-example-python/src/main/python/example/Book.py
from dataclasses import dataclass
from typing import Annotated

from micronaut.data.annotation import GeneratedValue, Id, MappedEntity, Relation
from micronaut.serde.annotation import Serdeable

from example.Author import Author


@Serdeable
@MappedEntity
@dataclass
class Book:
    title: str
    pages: int
    author: Annotated[Author | None, Relation("MANY_TO_ONE")] = None
    id: Annotated[int | None, Id, GeneratedValue] = None
